### PR TITLE
Update library template to be suitable for npm publish

### PR DESCRIPTION
**Pull request by msinkec**

### CODE REVIEW

**Comment by msinkec** [comment]
I've updated the the library template and made sure it works if you publish. You just run `npm publish`. I think most of the things I've wrote in the old CLI publish command are reduntant. If the user fiddles with build configs their sCrypt code likely won't build anyway and publishing will fail. In places where a particular config is needed for a valid publish (e.g. explicit inclusion of dist/ via .npmignore or output of type declarations via tsconfig.json) I've added comments on why you need them so users know what they're getting into if they delete something. I think this is good enough since everything will work by default unless the user deliberately wrecks something.

File: templates/counter/src/contracts/PROJECT_NAME.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
+{
```
[the remark this answers is not in the document]
Comment (msinkec) [reply templates/counter/src/contracts/PROJECT_NAME.json:1]: Hmm, these should not be there, no. Thanks for noticing. I see that they get generated while running `test-templates`. I see you've made some changes there. Can you check?

**Comment by xhliu** [comment]
Let's ship and update the doc.

File: templates/counter/src/contracts/PROJECT_NAME.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (zhfnjust) [reply templates/counter/src/contracts/PROJECT_NAME.json:1]: i check the latest master. it won't generate that json file now

[the remark this answers is not in the document]
Comment (zhfnjust) [reply templates/counter/src/contracts/PROJECT_NAME.json:1]: latest code put the json file in the `artifacts/` directory

[the remark this answers is not in the document]
Comment (msinkec) [reply templates/counter/src/contracts/PROJECT_NAME.json]: Seems like the issue was the `npx scryt-cli compile` command which used an older version. I now changed it to invoke `index.js` directly.

**Pull request merged**